### PR TITLE
AGENT-387 port luajson to the linux kernel

### PR DESCRIPTION
**Pull request by tammela**

### CODE REVIEW

File: luajson.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+	original  = *p;
+	*p = '\0';
+
+	if ((err = kstrtol(s, base, &res)) != 0)
+		res = err;
+
+	*p = original;
+
+	return res;
```
Comment (lneto) [inline luajson.h]: it could return garbage.. I would return 0 here and res in the if block..

File: luajson.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#ifndef LUAJSON_H
+#define LUAJSON_H
+
+#ifdef _KERNEL
+inline static long
+strtol (char *s, void *unused, int base)
+{
+	long res;
+	int err;
+	char original;
```
Comment (lneto) [inline luajson.h]: s/original/delim/

File: luajson.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#ifndef LUAJSON_H
+#define LUAJSON_H
+
+#ifdef _KERNEL
+inline static long
+strtol (char *s, void *unused, int base)
```
Comment (lneto) [inline luajson.h]: (void)unused to avoid warns..

File: luajson.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+{
+	long res;
+	int err;
+	char original;
+	char *p = strpbrk(s, " ,");
+
+	original  = *p;
```
Comment (lneto) [inline luajson.h]: s/ //

File: luajson.c [hunk 5]
```diff
@@ -364,9 +338,13 @@ decode_value(lua_State *L, char **s, int null)
 			(*s)++;
 		}
 		if (isfloat)
-			lua_pushnumber(L, atof(p));
+#ifndef _KERNEL
+			lua_pushnumber(L, strtod(p, NULL));
+#else
+			luaL_error(L, "floating point numbers are not supported.");
```
Comment (lneto) [inline luajson.c]: I would remove float cases from the while above and let it fail with a syntax error..

File: luajson.c [hunk 6]
```diff
@@ -36,12 +37,21 @@
...
 #include <stdarg.h>
+#ifndef _KERNEL
 #include <stdio.h>
 #include <string.h>
 #include <ctype.h>
 #include <stdlib.h>
+#endif
+
+#ifdef _KERNEL
+#define malloc(sz) kmalloc(sz, GFP_KERNEL)
```
Comment (tammela) [inline luajson.c]: free is defined in luaconf.h, lua doesn't use malloc, only realloc

File: luajson.h [hunk 7]
```diff
@@ -0,0 +1,27 @@
+#ifndef LUAJSON_H
+#define LUAJSON_H
+
+#ifdef _KERNEL
+inline static long
+strtol (char *s, void *unused, int base)
+{
+	long res;
+	int err;
```
Comment (tammela) [inline luajson.h]: ?

**Pull request closed without merging**